src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardSetAddOrRemoveByContainsTests.cs

```diff
@@ -0,0 +1,841 @@
+                MySet.Remove(""Item"");
+            else
+                throw new Exception(""Item doesn't exist"");
+        }" + CSNamespaceAndClassEnd;
+
+            string fixedSource = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+
+        public MyClass()
+        {
+            if (!MySet.Remove(""Item""))
+                throw new Exception(""Item doesn't exist"");
+        }" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task AddHasElseBlock_OffersFixer_CS()
+        {
+            string source = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+
+        public MyClass()
+        {
+            if (![|MySet.Contains(""Item"")|])
+            {
+                MySet.Add(""Item"");
+            }
+            else
+            {
+                throw new Exception(""Item already exists"");
+            }
+        }" + CSNamespaceAndClassEnd;
+
+            string fixedSource = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+
+        public MyClass()
+        {
+            if (!MySet.Add(""Item""))
+            {
+                throw new Exception(""Item already exists"");
+            }
+        }" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task RemoveHasElseBlock_OffersFixer_CS()
+        {
+            string source = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+
+        public MyClass()
+        {
+            if ([|MySet.Contains(""Item"")|])
+            {
+                MySet.Remove(""Item"");
+            }
+            else
+            {
+                throw new Exception(""Item doesn't exist"");
+            }
+        }" + CSNamespaceAndClassEnd;
+
+            string fixedSource = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+
+        public MyClass()
+        {
+            if (!MySet.Remove(""Item""))
+            {
+                throw new Exception(""Item doesn't exist"");
+            }
+        }" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task AddWithAdditionalStatements_ReportsDiagnostic_CS()
+        {
+            string source = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+
+        public MyClass()
+        {
+            if (![|MySet.Contains(""Item"")|])
+            {
+                MySet.Add(""Item"");
+                Console.WriteLine();
+            }
```

Comment:
Simple additional statements like this could have code fix suggestions, but it is not required for this PR, we can file an issue for further improvement